subsys/shell/shell_uart.c

```diff
@@ -165,6 +171,11 @@ static int init(const struct shell_transport *transport,
 	sh_uart->ctrl_blk->handler = evt_handler;
 	sh_uart->ctrl_blk->context = context;
 
+#ifdef CONFIG_MCUMGR_SMP_SHELL
```

Comment:
if you would use `IF_ENABLED(CONFIG_MCUMGR_SMP_SHELL)` then ifdef around pool definition would not be needed and code would be cleaner.

Reply:
This won't work:
```
	if (IS_ENABLED(CONFIG_MCUMGR_SMP_SHELL)) {
		sh_uart->ctrl_blk->smp.buf_pool = &smp_shell_rx_pool;
		k_fifo_init(&sh_uart->ctrl_blk->smp.buf_ready);
	}
```
because smp member will be missing in:
```
/** @brief Shell UART transport instance control block (RW data). */
struct shell_uart_ctrl_blk {
	struct device *dev;
	shell_transport_handler_t handler;
	void *context;
	atomic_t tx_busy;
	bool blocking_tx;
#ifdef CONFIG_MCUMGR_SMP_SHELL
	struct smp_shell_data smp;
#endif /* CONFIG_MCUMGR_SMP_SHELL */
};
```
resulting in build failures:
```
zephyr/subsys/shell/shell_uart.c:175:20: error: 'struct shell_uart_ctrl_blk' has no member named 'smp'
  175 |   sh_uart->ctrl_blk->smp.buf_pool = &smp_shell_rx_pool;
      |
```